R/ggplotMDS.R

```diff
@@ -96,105 +139,209 @@ ggplotMDS <- function(DGEdata,
+                    warning("reflineSize must be a singular numeric value greater than 0. Assigning default value '0.5'.")
+                    reflineSize <- 0.5
+                }
+            }
+        }
     }
-    if (!missing(sizeBy)) {
-        assertthat::assert_that(length(sizeBy) == ncol(DGEdata),
-                                msg = "sizeBy should be the length of the number of columns in DGEdata.")
+
+    if (plotType == "ggplot") {
+
+        if (!missing(labels)) {
+            if (any(!is.numeric(labelSize), length(labelSize) != 1, labelSize < 0)) {
+                warning("labelSize should be singular numeric value and greater than zero. Assigning default value 3.")
+                labelSize <- 3
+            }
+        }
+
+        intercept_flag <- FALSE
+        intercept_length <- 0
+        if (!missing(hlineIntercept)) {
+            if (!is.null(hlineIntercept)) {
+                intercept_flag <- TRUE
+                if (!is.numeric(hlineIntercept)) {
+                    warning("hlineIntercept must be numeric. Ignoring hlineIntercept.")
+                    hlineIntercept <- NULL
+                } else {
+                    intercept_length <- length(hlineIntercept)
+                }
+            }
+        }
+
+        if (!missing(vlineIntercept)) {
+            if (!is.null(vlineIntercept)) {
+                intercept_flag <- TRUE
+                if (!is.numeric(vlineIntercept)) {
+                    warning("vlineIntercept must be numeric. Ignoring vlineIntercept.")
+                    vlineIntercept <- NULL
+                } else {
+                    intercept_length <- length(vlineIntercept)
+                }
+            }
+        }
+
+        if (intercept_flag) {
+            if (!missing(reflineColor)) {
+                if (!is.null(reflineColor) & !is.character(reflineColor)) {
+                    warning("reflineColor must be a of class character and must specify the name of the color or the rgb value. Assigning default value 'red'.")
+                    reflineColor <- "red"
+                } else if (.rgbaConversion(reflineColor) == "invalid value") {
+                    warning("Color specified is not valid. Assigning default value 'red'.")
+                    reflineColor <- "red"
+                } else if (!(length(reflineColor) == intercept_length | length(reflineColor) == 1)) {
+                    warning("reflineColor must be either length 1 or the same as the intercept. Assigning default value 'red'.")
+                    reflineColor <- "red"
+                }
+            }
+
+            if (!missing(reflineSize)) {
+                if (any(is.null(reflineSize), !is.numeric(reflineSize), any(reflineSize < 0))) {
+                    warning("reflineSize must be a numeric value greater than 0. Assigning default value '0.5'.")
+                    reflineSize <- 0.5
+                }  else if (!(length(reflineSize) == intercept_length | length(reflineSize) == 1)) {
```

Comment:
* **||** instead of  **|**
* Why combining the conditions and negating the results with or, you can and the negative of these conditions which will enhance readability and maintainability